src/publish.rs

```diff
@@ -296,14 +297,14 @@ pub fn publish_crate(mut body: Bytes, index: &mut Index) -> Result<()> {
     .write(&data)
     .with_context(|| format!("failed to write to crate file {}", crate_path.display()))?;
 
-  index.add(&crate_meta_path).with_context(|| {
+  index.add(&crate_meta_relative_path).with_context(|| {
     format!(
       "failed to add {} to git repository",
       crate_meta_path.display()
     )
   })?;
   index
-    .add(&crate_path)
+    .add(&crate_relative_path)
     .with_context(|| format!("failed to add {} to git repository", crate_path.display()))?;
```

Comment:
```suggestion
    .with_context(|| format!("failed to add {} to git repository", crate_relative_path.display()))?;
```

Reply:
Well, I specifically left it so because it would be better to see the full path in the error message for debugging. Maybe display both?